src/db.rs

```diff
@@ -1180,7 +1186,10 @@ impl<T: ThreadMode, D: DBInner> DBCommon<T, D> {
     {
         let (keys, keys_sizes): (Vec<Box<[u8]>>, Vec<_>) = keys
             .into_iter()
-            .map(|k| (Box::from(k.as_ref()), k.as_ref().len()))
+            .map(|k| {
+                let k = k.as_ref();
+                (Box::from(k), k.len())
```

Comment:
```suggestion
                (k.as_ptr() as *const c_char, k.len())
```

Hijacking this PR to point out a useless allocation around these changes: I believe the intermediary `Box` here and in all similar `multi_get` methods is unnecessary and we could simply collect the keys directly into `Vec<*const _c_char>`?